lib/diff.ml

```diff
@@ -38,18 +38,28 @@ and sig_item =
 let extract_items items =
   List.fold_left
     (fun tbl item ->
+      let visibility = Types.item_visibility item in
       match item with
       | Sig_module (id, _, mod_decl, _, _) ->
-          Sig_item_map.add ~name:(Ident.name id) Sig_item_map.Module mod_decl
-            tbl
+          if visibility = Types.Hidden then tbl
```

Comment:
Same here, you don't need to go through an if-then-else, you can directly match over the `visibility` argument (last one) and let the pattern matching wildcard case at the end handle all `Hidden` items.